### DISCUSSION TITLE
Possible to scrub PII from what gets logged to langsmith, while still using unredacted data in the actual LLM call?

### DISCUSSION THREAD

**Discussion by phoebe-ross** [langchain-ai/langsmith-sdk#579] [Polls]
- Langsmith's TOC prohibits PII from being logged to its service (more stringent than LLM providers)
- The `process_inputs` argument for @traceable can be used to filter PII from the inputs that are logged to langsmith, but it does not look like there is an analogous way for logged output (is this true?)
- I think _setup_run and _container_end can be patched to add pii scrubbing, but this doesn't feel like a great solution

**Comment by hinthornw** [comment]
Does this address your question? 

https://docs.smith.langchain.com/tracing/faq/customizing_trace_attributes#masking-inputs-and-outputs
